src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelGetSemanticInfoTests.cs

```diff
@@ -5553,6 +5555,54 @@ void M()
             Assert.True(semanticInfo.Symbol.GetSymbol().IsFromCompilation(compilation));
         }
 
+        [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+        [Fact]
+        public void LocalForEach()
+        {
+            string sourceCode = @"
+class C
+{
+    void M()
+    {
+        var a = new int[] { 1, 2, 3 };
+        foreach (var x in a)
+        {
+            var y = /*<bind>*/x/*</bind>*/;
+        }
+    }
+}
+";
+            var compilation = CreateCompilation(sourceCode);
+            var semanticInfo = GetSemanticInfoForTest<IdentifierNameSyntax>(compilation);
+
+            var symbol = (ILocalSymbol)semanticInfo.Symbol;
+            Assert.True(symbol.IsForEach);
+        }
+
+        [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+        [Fact]
+        public void LocalUsing()
+        {
+            string sourceCode = @"
+using System.IO;
+class C
+{
+    void M()
+    {
+        using (var x = new StreamReader(""C:\\Temp\\MyFile.txt""))
```

Comment:
Let's add a test for `using var = ...;` as well.

Reply:
f6b7e73e093cd08e788dab17a2f8fb2f50ef713f